### PR TITLE
refactor multivalue fastfield, refactor range query

### PR DESCRIPTION
**Pull request by PSeitz**
Introduce MakeZero trait, remove make_zero from FastValue
Merge two multivalue fastfield implementations into one
prepare range query on fastfield for different types

### CODE REVIEW

File: src/query/range_query/fast_field_range_query.rs [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+    }
+    fn next(&mut self) -> Option<u32> {
+        self.current_pos += 1;
+        self.current()
+    }
+    #[inline]
+    fn current(&self) -> Option<u32> {
+        self.docs.get(self.current_pos).copied()
+    }
+    fn get_cleared_data(&mut self) -> &mut Vec<u32> {
```
Comment (fulmicoton) [inline src/query/range_query/fast_field_range_query.rs:29]: That's a very strange API.
I'm not sure where you are going with this.

Could we use `Into<Vec<u32>>` instead?

File: src/query/range_query/fast_field_range_query.rs [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+        self.docs.clear();
+        self.current_pos = 0;
+        &mut self.docs
+    }
+    fn last_value(&self) -> Option<u32> {
+        self.docs.iter().last().cloned()
+    }
+    fn is_empty(&self) -> bool {
+        self.current_pos >= self.docs.len()
```
Comment (fulmicoton) [inline src/query/range_query/fast_field_range_query.rs]: alternatively `self.current.is_none()`...
there are pros and cons to both.

File: src/query/range_query/fast_field_range_query.rs [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+        while doc < target {
+            doc = self.advance();
+        }
+        self.last_seek_pos_opt = Some(target);
+        doc
+    }
+
+    fn size_hint(&self) -> u32 {
+        0 // heuristic possible by checking number of hits when fetching a block
+    }
```
Comment (fulmicoton) [inline src/query/range_query/fast_field_range_query.rs:208]: can you add unit tests on this module? I think there are a lot of tricky logic in this

**Review by fulmicoton — APPROVED** [review]

File: src/query/range_query/fast_field_range_query.rs (hunk 2 above)
Comment (PSeitz) [reply src/query/range_query/fast_field_range_query.rs]: I like yours better

File: src/query/range_query/fast_field_range_query.rs (hunk 3 above)
Comment (PSeitz) [reply src/query/range_query/fast_field_range_query.rs:208]: There are tests in `range_query_ip_fastfield.rs`. I will add another PR that enables range queries on fastfield for all types. I'll try to cover all branches with tests there

**Pull request merged**